packages/cardpay-sdk/sdk/reward-pool/base.ts

```diff
@@ -93,35 +97,12 @@ export default class RewardPool {
     const tokenContract = new this.layer2Web3.eth.Contract(ERC20ABI as AbiItem[], tokenAddress);
     return await tokenContract.methods.balanceOf(rewardPoolAddress).call();
   }
-
-  async rewardTokensAvailable(rewardProgramId?: string, address?: string): Promise<string[]> {
-    let tallyServiceURL = await getConstant('tallyServiceURL', this.layer2Web3);
-    let url = new URL(`${tallyServiceURL}/reward-tokens/`);
-    if (rewardProgramId) {
-      url.searchParams.append('reward_program_id', rewardProgramId);
-    }
-    if (address) {
-      url.searchParams.append('payee', address);
-    }
-    let options = {
-      method: 'GET',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-    };
-    let response = await fetch(url.toString(), options);
-    if (!response?.ok) {
-      throw new Error(await response.text());
-    }
-    let json = await response.json();
-    return json['tokenAddresses'];
-  }
-
   async getProofs(
     address: string,
     rewardProgramId?: string,
     tokenAddress?: string,
     knownClaimed?: boolean,
+    safeAddress?: string,
```

Comment:
There was some discussion of using `override` type for function and extend the `Proof` type. I intentionally found that using `optional` parameters much easier and straightforward. `type` based switch case don't seem too straightforward and I still need to use it in other functions -- this solution is very easy to grok despite needing to check existence of parameter `safeAddress` in multiple places.

Reply:
EDIT: Ended up introducing new interface `ClaimableProof` and using override. It was useful after ian told me to use `proofs` as the only input to `claimAll` and `claimAllGasEstimates`